functions.php

```diff
@@ -128,32 +128,25 @@ function panel_login_fail( $username ) {
     }
 }
 
-function idi_display_twitter_feed($twitter_un) {
-    require_once("wp-content/plugins/twitteroauth/twitteroauth/twitteroauth.php");
-    $num_tweets = 1;
-    $consumerkey = "luK78NyRjDEmMVhi6sgIw";
-    $consumersecret = "E6bY0ShFmtibIqWU0oHokCVZKYtPEvZcNyACBPzYqo";
-    $accesstoken = "123905660-3gtwAKtHHrPjGwa1PmAqXD8FKKKQY2C1ORB8dpyE";
-    $accesstokensecret = "kWuqrTk8IOHVSQdEK9dFfy337VjQv9P44lYKVfq8Fv7Ln";
-    $connection = new TwitterOAuth($consumerkey, $consumersecret, $accesstoken, $accesstokensecret);
-    $tweets = array_filter($connection->get("https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=".$twitter_un."&count=".$num_tweets));
-
-	echo '<div class="idi-box idi-highlight-box twitter-feed-group">
-				<div class="idi-box-text">
-					<a class="twitter-follow-button" rel="external nofollow" href="http://twitter.com/'.$twitter_un.'" title="Follow @'.$twitter_un.'">@'.$twitter_un.'</a>
-					<ul>
-	';
-	if(!empty($tweets)){
-	        foreach($tweets as $tweet) {
-                echo '<li class="tweet">'.$tweet->text.
-                '<div class="tweet-date">'.substr($tweet->created_at, 0, 16).'</div></li>';
+function idi_display_twitter_feed($twitter_username) {
+
+    /* Use the "oAuth Twitter for Developers" plugin to get the Twitter feed.
+       https://en-ca.wordpress.org/plugins/oauth-twitter-feed-for-developers/ */
+    if (function_exists('getTweets')) {
+        $tweet_count = 5;
+        $tweets = getTweets($tweet_count, $twitter_username);
+        $out = '<div class="idi-box idi-highlight-box twitter-feed-group"><div class="idi-box-text"><a class="twitter-follow-button" rel="external nofollow" href="http://twitter.com/';
+        $out .= $twitter_username.'" title="Follow @'.$twitter_username.'">@'.$twitter_username.'</a><ul>';
+        if (!empty($tweets)) {
+            foreach ($tweets as $tweet) {
+                $out .= '<li class="tweet"><div class="tweet-date">'.substr($tweet[created_at], 0, 16).'</div>'.$tweet[text].'</li>';
```

Comment:
If the twitter plugin is installed but the consumer key and secret is not provided, this error will appear a number of times on the front page:

`Warning: Illegal string offset 'created_at' in /.../wp-content/themes/idi-theme/functions.php on line 142`